src/js/ripple/orderbook.js

```diff
@@ -922,41 +927,59 @@ OrderBook.prototype.insertOffer = function(node) {
 
 OrderBook.prototype.modifyOffer = function(node) {
   if (this._remote.trace) {
-    if (node.nodeType === 'DeletedNode') {
-      log.info('deleting offer', this._key, node.fields);
-    } else {
-      log.info('modifying offer', this._key, node.fields);
+    log.info('modifying offer', this._key, node.fields);
+  }
+
+  for (var i=0; i<this._offers.length; i++) {
+    var offer = this._offers[i];
+
+    if (offer.index === node.ledgerIndex) {
+      // TODO: This assumes no fields are deleted, which is
+      // probably a safe assumption, but should be checked.
+      extend(offer, node.fieldsFinal);
+
+      break;
     }
   }
 
-  for (var i=0, l=this._offers.length; i<l; i++) {
+  this.updateOwnerOffersFundedAmount(node.fields.Account);
+};
+
+/** 
+ * Delete an existing offer in the orderbook
+ *
+ * NOTE: We only update funded amounts when the node comes from an OfferCancel
+ *       transaction because when offers are deleted, it frees up funds to fund
+ *       other existing offers in the book
+ *
+ * @param {Object} node - Offer node
+ * @param {Boolean} isOfferCancel - whether node came from an OfferCancel transaction
+ */
+
+OrderBook.prototype.deleteOffer = function(node, isOfferCancel) {
+  if (this._remote.trace) {
+    log.info('deleting offer', this._key, node.fields);
+  }
+
+  for (var i=0; i<this._offers.length; i++) {
```

Comment:
Unfortunately we're doing concurrent modification here so I didn't want to use any iteration HOFs